#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "../protos/Pickup_robot.proto"
EXTERNPROTO "../protos/LED_Richting.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.28168711819333275 -0.5172996055946125 0.8081172473692128 5.097304182941846
  position -0.5690123327901073 1.944356624552609 1.4790431310714112
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0.45 0.45 0
  floorTileSize 0.2 0.2
}
DEF PR_1_LED_0 LED_Richting {
  translation 0.8999999999999999 0.7599999999999999 0.05
  name " PR_0_LED_0"
}
DEF PR_0_LED_0 LED_Richting {
  translation 0.8999999999999999 0.8599999999999999 0.05
  name " PR_0_LED_0(1)"
}
DEF PR_1_LED_2 LED_Richting {
  translation 0.8999999999999999 0.84 0.05
  name " PR_0_LED_2"
}
DEF PR_0_LED_2 LED_Richting {
  translation 0.8999999999999999 0.94 0.05
  name " PR_0_LED_2(1)"
}
DEF PR_1_LED_1 LED_Richting {
  translation 0.8599999999999999 0.7999999999999999 0.05
  name " PR_0_LED_1"
}
DEF PR_0_LED_1 LED_Richting {
  translation 0.8599999999999999 0.8999999999999999 0.05
  name " PR_0_LED_1(1)"
}
DEF PR_1_LED_3 LED_Richting {
  translation 0.94 0.7999999999999999 0.05
  name " PR_0_LED_3"
}
DEF PR_0_LED_3 LED_Richting {
  translation 0.94 0.8999999999999999 0.05
  name " PR_0_LED_3(1)"
}
DEF pick_up_robot_1 Pickup_robot {
  name "pick_up_robot_1"
  translation 0.8999999999999999 0.7999999999999999 0
  target 0 1
  controller "pickup_robot_1"
  children [
    DEF PR_1_DS_0 DistanceSensor {
      translation 4.07848e-08 -0.04 -0.38
      rotation 0 0 1 -1.5707953071795862
      name "PR_1_DS_0"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_1_DS_1 DistanceSensor {
      translation -0.04 0 -0.38
      rotation 0 0 1 3.14159
      name "PR_1_DS_1"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_1_DS_2 DistanceSensor {
      translation -1.46928e-07 0.04 -0.38
      rotation 0 0 1 1.5708
      name "PR_1_DS_2"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_1_DS_3 DistanceSensor {
      translation 0.04 0 -0.38
      name "PR_1_DS_3"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
  ]
  sensors [
    1000, 111, 1000, 1000
  ]
  emergency TRUE
}
DEF Centrale_server Robot {
  translation 0 0 0.63
  children [
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0.333333 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
    }
  ]
  name "centrale_server"
  controller "Central_server_py"
  supervisor TRUE
}
DEF pick_up_robot_0 Pickup_robot {
  name "pick_up_robot_0"
  color 0.666667 0.666667 1
  translation 0.8999999999999999 0.8999999999999999 0
  target 0 0
  children [
    DEF PR_0_DS_0 DistanceSensor {
      translation 4.07848e-08 -0.04 -0.38
      rotation 0 0 1 -1.5707953071795862
      name "PR_0_DS_0"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_0_DS_1 DistanceSensor {
      translation -0.04 0 -0.38
      rotation 0 0 1 3.14159
      name "PR_0_DS_1"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_0_DS_2 DistanceSensor {
      translation -1.46928e-07 0.04 -0.38
      rotation 0 0 1 1.5708
      name "PR_0_DS_2"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_0_DS_3 DistanceSensor {
      translation 0.04 0 -0.38
      name "PR_0_DS_3"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
  ]
  sensors [
    1000, 111, 1000, 1000
  ]
  emergency TRUE
}
DEF pick_up_robot_2 Pickup_robot {
  name "pick_up_robot_2"
  color 0.956863 1 0.160784
  translation 0.5 2.7755575615628914e-17 0
  target 0 9
  controller "pickup_robot_2"
  children [
    DEF PR_2_DS_0 DistanceSensor {
      translation 4.07848e-08 -0.04 -0.38
      rotation 0 0 1 -1.5707953071795862
      name "PR_2_DS_0"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_2_DS_1 DistanceSensor {
      translation -0.04 0 -0.38
      rotation 0 0 1 3.14159
      name "PR_2_DS_1"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_2_DS_2 DistanceSensor {
      translation -1.46928e-07 0.04 -0.38
      rotation 0 0 1 1.5708
      name "PR_2_DS_2"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_2_DS_3 DistanceSensor {
      translation 0.04 0 -0.38
      name "PR_2_DS_3"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
  ]
  sensors [
    1000, 111, 111, 1000
  ]
  emergency TRUE
}
DEF PR_2_LED_0 LED_Richting {
  translation 0.5 -0.03999999999999997 0.05
  name "PR_2_LED_0"
}
DEF PR_2_LED_1 LED_Richting {
  translation 0.46 2.7755575615628914e-17 0.05
  name "PR_2_LED_1"
}
DEF PR_2_LED_2 LED_Richting {
  translation 0.5 0.04000000000000003 0.05
  name "PR_2_LED_2"
}
DEF PR_2_LED_3 LED_Richting {
  translation 0.54 2.7755575615628914e-17 0.05
  name "PR_2_LED_3"
}
DEF pick_up_robot_3 Pickup_robot {
  name "pick_up_robot_3"
  color 0 1 0.498039
  translation 0.6 0.20000000000000004 0
  target 9 9
  controller "pickup_robot_3"
  children [
    DEF PR_3_DS_0 DistanceSensor {
      translation 4.07848e-08 -0.04 -0.38
      rotation 0 0 1 -1.5707953071795862
      name "PR_3_DS_0"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_3_DS_1 DistanceSensor {
      translation -0.04 0 -0.38
      rotation 0 0 1 3.14159
      name "PR_3_DS_1"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_3_DS_2 DistanceSensor {
      translation -1.46928e-07 0.04 -0.38
      rotation 0 0 1 1.5708
      name "PR_3_DS_2"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
    DEF PR_3_DS_3 DistanceSensor {
      translation 0.04 0 -0.38
      name "PR_3_DS_3"
      lookupTable [
        0 0 0
        0.09 1000 0
      ]
    }
  ]
  sensors [
    1000, 1000, 111, 111
  ]
  emergency TRUE
}
DEF PR_3_LED_0 LED_Richting {
  translation 0.6 0.16000000000000003 0.05
  name "PR_3_LED_0"
}
DEF PR_3_LED_1 LED_Richting {
  translation 0.5599999999999999 0.20000000000000004 0.05
  name "PR_3_LED_0(1)"
}
DEF PR_3_LED_2 LED_Richting {
  translation 0.6 0.24000000000000005 0.05
  name "PR_3_LED_0(2)"
}
DEF PR_3_LED_3 LED_Richting {
  translation 0.64 0.20000000000000004 0.05
  name "PR_3_LED_0(3)"
}
